"Error: Type mismatch in argument ..." when compiling with GPU support

I am trying to compile spral with GPU support, but get the following error after running the make command:
```
gfortran -fopenmp -fPIC -c -o src/ssids/gpu/factor.o src/ssids/gpu/factor.f90
src/ssids/gpu/factor.f90:2627:45:

 2627 |             gwork, st, cuda_error, ptr_scale)
      |                                             1
Error: Type mismatch in argument 'level_size' at (1); passed INTEGER(4) to INTEGER(8)
src/ssids/gpu/factor.f90:2631:34:

 2631 |             gwork, st, cuda_error)
      |                                  1
Error: Type mismatch in argument 'level_size' at (1); passed INTEGER(4) to INTEGER(8)
src/ssids/gpu/factor.f90:2525:24:

 2525 |          st, cuda_error)
      |                        1
Error: Type mismatch in argument 'off_ldlt' at (1); passed INTEGER(4) to INTEGER(8)
src/ssids/gpu/factor.f90:2259:26:

 2259 |          stats%cuda_error)
      |                          1
Error: Type mismatch in argument 'off_ldlt' at (1); passed INTEGER(4) to INTEGER(8)
src/ssids/gpu/factor.f90:2156:46:

 2156 |             gwork, stats%st, stats%cuda_error)
      |                                              1
Error: Type mismatch in argument 'pc_size' at (1); passed INTEGER(4) to INTEGER(8)
src/ssids/gpu/factor.f90:2156:46:

 2156 |             gwork, stats%st, stats%cuda_error)
      |                                              1
Error: Type mismatch in argument 'off_ldlt' at (1); passed INTEGER(4) to INTEGER(8)
make[1]: *** [Makefile:2513: src/ssids/gpu/factor.o] Error 1
make[1]: Leaving directory '/c/GBW_MyPrograms/coin-or/spral/source/spral'
make: *** [Makefile:1661: all] Error 2
```
I am using mingw-w64 on windows 10
The same error occurs when using blas and lapack from mkl
```
CFLAGS=-fPIC CPPFLAGS=-fPIC CXXFLAGS=-fPIC FFLAGS="-fPIC -fallow-argument-mismatch" \
   FCFLAGS="-fPIC -fallow-argument-mismatch" NVCCFLAGS="-shared -Xcompiler -fPIC" \
   ./configure --prefix="c:/GBW_MyPrograms/coin-or/spral/build/with-cuda-v1" \
   --with-blas="-lmkl_intel_lp64 -lmkl_sequential -lmkl_core" --with-lapack="-lmkl_intel_lp64 -lmkl_sequential -lmkl_core" \
   --with-metis="-lmetis"
```
or when getting them through mingw
```
pacman -S mingw-w64-x86_64-lapack mingw-w64-x86_64-metis  mingw-w64-x86_64-openblas
CFLAGS=-fPIC CPPFLAGS=-fPIC CXXFLAGS=-fPIC FFLAGS=-fPIC \
   FCFLAGS=-fPIC NVCCFLAGS="-shared -Xcompiler -fPIC" \
   ./configure --prefix="c:/GBW_MyPrograms/coin-or/spral/build/with-cuda-v1" \
   --with-blas="-lopenblas" --with-lapack="-llapack" \
   --with-metis="-lmetis"
```

Adding `-fallow-argument-mismatch` to the FFLAGS and FCFLAGS resulted in the same error.

Do you have any suggestions on what is going wrong?

--- a/examples/Fortran/ssids.f90
+++ b/examples/Fortran/ssids.f90
@@ -3,7 +3,7 @@ program ssids_example
    use spral_ssids
    implicit none
 
-   integer, parameter :: long = selected_int_kind(16)
+   integer, parameter :: long = selected_int_kind(18)
 
    ! Derived types
    type (ssids_akeep)   :: akeep

--- a/src/ssids/gpu/factor.f90
+++ b/src/ssids/gpu/factor.f90
@@ -2045,9 +2045,9 @@ contains
     type(c_ptr), value, intent(in) :: c_rptr
     type(c_ptr), value, intent(in) :: c_sptr
     type(c_ptr), value, intent(in) :: c_asminf
-    integer(c_long), value, intent(in) :: pc_size
+    integer(c_int64_t), value, intent(in) :: pc_size
     ! type(c_ptr), value, intent(in) :: c_off_LDLT
-    integer(c_long), dimension(*), intent(in) :: off_LDLT
+    integer(c_int64_t), dimension(*), intent(in) :: off_LDLT
     type(c_ptr), value, intent(in) :: ptr_ccval ! GPU (*ptr_ccval) points to previous
       ! level's contribution blocks
     type(c_ptr), value, intent(in) :: c_gpu_contribs
@@ -2181,7 +2181,7 @@ contains
     type(c_ptr), value, intent(in) :: c_nodes
     type(c_ptr), value, intent(in) :: c_rptr
     type(c_ptr), value, intent(in) :: c_sptr
-    integer(c_long), dimension(*), intent(in) :: off_LDLT
+    integer(c_int64_t), dimension(*), intent(in) :: off_LDLT
     ! type(c_ptr), value, intent(in) :: c_off_LDLT
     type(c_ptr), value             :: ptr_levLDLT 
     type(c_ptr), value             :: c_gwork ! GPU workspace allocator 
@@ -2407,7 +2407,7 @@ contains
     type(c_ptr), value, intent(in) :: c_child_ptr 
     type(c_ptr), value, intent(in) :: c_child_list 
     ! type(c_ptr), value, intent(in) :: c_off_LDLT
-    integer(c_long), dimension(*) :: off_LDLT
+    integer(c_int64_t), dimension(*) :: off_LDLT
     type(c_ptr), value, intent(in) :: c_asminf ! Assembly info 
     type(c_ptr), value, intent(in) :: c_rptr
     type(c_ptr), value, intent(in) :: c_sptr
@@ -2550,7 +2550,7 @@ contains
     type(c_ptr), value, intent(in) :: c_lvllist
     type(c_ptr), value, intent(in) :: c_nodes
     integer(c_int), value :: ncb ! Number of nodes in level
-    integer(c_long), value :: level_size ! Number of (fully-summed) entries in level  
+    integer(c_int64_t), value :: level_size ! Number of (fully-summed) entries in level  
     type(c_ptr), value, intent(in) :: c_nptr
     type(c_ptr), value, intent(in) :: c_rptr
     type(c_ptr), value, intent(in) :: gpu_nlist ! nlist array on the GPU

